src/gpu/d3d11/SDL_gpu_d3d11.c

```diff
@@ -5240,32 +5258,41 @@ static bool D3D11_INTERNAL_CreateSwapchain(
 
 static bool D3D11_INTERNAL_ResizeSwapchain(
     D3D11Renderer *renderer,
-    D3D11WindowData *windowData,
-    Sint32 width,
-    Sint32 height)
+    D3D11WindowData *windowData)
 {
+    D3D11_Wait((SDL_GPURenderer *)renderer);
+
     // Release the old RTV
     ID3D11RenderTargetView_Release(windowData->texture.subresources[0].colorTargetViews[0]);
     SDL_free(windowData->texture.subresources[0].colorTargetViews);
     SDL_free(windowData->texture.subresources);
 
+    int w, h;
+    SDL_SyncWindow(windowData->window);
```

Comment:
This function can only be called on the main thread. Are we guaranteed that both call sites are on the main thread?

Reply:
No, this function could be called from a thread if the window was created on that thread. How can we address this?

Reply:
Sorry, to be clear, it should only be called on the thread that created the window. I think other places in SDL assume that that's always the main thread, but that's outside the scope of this PR. If you are creating the window and resizing the swap chain all on a single thread, then I think you're also getting events on that thread, and you don't need any locking. Can you verify that?

Reply:
We explicitly specify that AcquireGPUSwapchainTexture can only be called from the thread that created the window, and it's a spec violation to access command buffers from multiple threads, so that should be fine. I'll try to write a quick smoke test to check the event thread.

Reply:
If that's the case, we don't even need an atomic int there.

Reply:
Alright, reverting all of that then.